File: API/DrivingSchool.Core/Models/Student.cs
﻿namespace DrivingSchool.Core.Models
{
    public class Student
    {
        public int Id { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public DateTime YearOfBirth { get; set; }
        public string Email { get; set; }
        public string MobilePhone { get; set; }
        public string Address { get; set; }
        public string City { get; set; }
        public string TrainingCategory { get; set; }
        public bool IsTheory { get; set; }
        public bool IsPractical { get; set; }
        public DateTime? ExamTime { get; set; }
        public DateTime RegistrationDate { get; set; } = DateTime.Today;
        public string UniqueId { get; set; } = Guid.NewGuid().ToString();
        public bool EmailSent { get; set; } = false;
    }
}

File: drive-school-react/src/types/student.ts
export type Student = {
    firstName: string;
    lastName: string;
    yearOfBirth: Date;
    email: string;
    mobilePhone: string;
    address: string;
    city: string;
    trainingCategory: string;
    isTheory: boolean;
    isPractical: boolean;
    examTime: Date | null;
}